### ISSUE TITLE
No way to hide menu bar icon

### ISSUE THREAD

**Issue by YodaLightsabr** [docker/roadmap#486]
There is currently no way to hide the Docker menu bar icon, as noted in docker/for-mac#2272 before it was closed for inactivity. This is an annoyance for many users, and it violates Apple's Human Interface Guidelines.

  - [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: `DFDF63AE-9730-428B-A471-285C1F9C8C86/20230521213534`
 

### Expected behavior
Docker allows users to hide the menu bar icon.

### Actual behavior
Docker prevents users from hiding the menu bar icon.

### Information

  - macOS Version: 13.0
  - Intel chip or Apple chip: Apple M2
  - Docker Desktop Version: 4.19.0

### Steps to reproduce the behavior

  1. Open Docker

**Comment by YodaLightsabr** [comment]
I can see how this could also be considered a feature request, but I think diverging from Apple's Human Interface Guidelines in a significant way that impacts a large share of users is a bug that should not be overlooked.

**Comment by gomsim** [comment]
> I can see how this could also be considered a feature request, but I think blatantly violating Apple's Human Interface Guidelines in away that annoys a large share of users is a bug that should not be overlooked.

Yup! From Human interface guideline's menu bar section (https://developer.apple.com/design/human-interface-guidelines/the-menu-bar):

_Let people — not your app — decide whether to put your menu bar extra in the menu bar. Typically, people add a menu bar extra to the menu bar by changing a setting in an app’s settings window. To ensure discoverability, however, consider giving people the option of doing so during setup._

**Comment by landsman** [comment]
I want to hide it too

**Comment by r4zzz4k** [comment]
Seems to be a duplicate of #281.

I'm also hoping to see this implemented, as the space in menu bar is priceless, and Docker Engine can nowadays be controlled via the status bar within the app, so there seem to be no unique functionality to the menu bar icon anyway.

**Comment by antonsacred** [comment]
I vote for this bug/feature to be solved

**Comment by Chrisser73** [comment]
pls implement, thx!

**Comment by ichmowgli** [comment]
In this case I’m pretty grateful the Bartender app exists... Still it would be nice if Docker had a built-in option to hide menu bar icons directly in settings w/o needing a third-party app

**Comment by rikur** [comment]
This is needed
